app/src/test/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlayTest.kt

```diff
@@ -30,74 +46,145 @@ class TrackingProtectionOverlayTest {
     @MockK(relaxed = true) private lateinit var metrics: MetricController
     @MockK(relaxed = true) private lateinit var toolbar: View
     @MockK(relaxed = true) private lateinit var icon: View
-    @MockK(relaxed = true) private lateinit var session: Session
+    @MockK(relaxed = true) private lateinit var session: SessionState
     @MockK(relaxed = true) private lateinit var overlay: TrackingProtectionOverlay
 
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule(testDispatcher)
+    private lateinit var store: BrowserStore
+
     @Before
     fun setup() {
         MockKAnnotations.init(this)
         context = spyk(testContext)
-
-        overlay = TrackingProtectionOverlay(context, settings, metrics) { toolbar }
+        store = BrowserStore()
+        val lifecycleOwner = MockedLifecycleOwner(Lifecycle.State.STARTED)
+
+        overlay = spyk(
+            TrackingProtectionOverlay(
+                context,
+                settings,
+                metrics,
+                store,
+                lifecycleOwner
+            ) { toolbar })
         every { toolbar.findViewById<View>(R.id.mozac_browser_toolbar_tracking_protection_indicator) } returns icon
     }
 
+    @After
+    fun cleanUp() {
+        testDispatcher.cleanupTestCoroutines()
+    }
+
+    @Test
+    fun `WHEN the loadingTHEN the overly is notified`() {
```

Comment:
nit: test name, maybe: WHEN loading state changes THEN overlay is notified....

Reply:
👍🏽